hack/code/prices_gen.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws/session"
+	ec22 "github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/karpenter/pkg/cloudprovider/aws"
+	"go/format"
+	"io/ioutil"
+	"log"
+	"os"
+	"runtime"
+	"runtime/pprof"
+	"sort"
+	"strings"
+	"time"
+)
+
+func main() {
+	flag.Parse()
+	if flag.NArg() != 1 {
+		log.Printf("Usage: %s -- pkg/cloudprovider/aws/zz_generated.pricing.go", os.Args[0])
+		os.Exit(1)
+	}
+
+	f, err := os.Create("pricing.heapprofile")
+	if err != nil {
+		log.Fatal("could not create memory profile: ", err)
+	}
+	defer f.Close() // error handling omitted for example
+
+	const region = "us-east-1"
+	os.Setenv("AWS_SDK_LOAD_CONFIG", "true")
+	os.Setenv("AWS_REGION", region)
+	ctx := context.Background()
+	sess := session.Must(session.NewSession())
+	ec2 := ec22.New(sess)
+	pricingProvider := aws.NewPricingProvider(ctx, aws.NewPricingAPI(sess, region), ec2, region)
+
+	src := &bytes.Buffer{}
+	fmt.Fprintln(src, "//go:build !ignore_autogenerated")
+	fmt.Fprintln(src, "package aws")
+	fmt.Fprintln(src, `import "time"`)
+	now := time.Now().Format(time.RFC822Z)
+	fmt.Fprintf(src, "// generated at %s for %s\n\n\n", time.Now().Format(time.RFC822Z), region)
+	fmt.Fprintf(src, "var initialPriceUpdate, _ = time.Parse(time.RFC822Z, \"%s\")\n", now)
+
+	instanceTypes := pricingProvider.InstanceTypes()
+	sort.Strings(instanceTypes)
+
+	writePricing(src, instanceTypes, "initialOnDemandPrices", pricingProvider.OnDemandPrice)
```

Comment:
We could simply hydrate the maps and then dynamically update it, rather than maintaining a concept of initial vs updated.
```suggestion
	writePricing(src, instanceTypes, "DemandPrices", pricingProvider.OnDemandPrice)
```

Reply:
I didn't want the static data exposed, I track the staleness time intentionally as well.  I plan on using it in a PR at some point when launching spot instances so we don't launch a spot that costs more than the cheapest on-demand.

Reply:
>I didn't want the static data exposed

Didn't mean to imply this should be public.